### PR TITLE
Add api-key proxy package (#253)

### LINKED ISSUE #253: [api-key] internal/auth_methods/api_key proxy package
Part of #243 — see that issue for full project context.

## Scope

New `internal/auth_methods/api_key/` package implementing `iface.Proxy`, mirroring the structure of `internal/auth_methods/oauth2/` and `internal/auth_methods/no_auth/`.

### Behavior

1. Load the active `api_key_credentials` row for the connection.
2. Decrypt via the existing encryption service.
3. Inject per `placement_snapshot`:
   - `bearer` → `Authorization: Bearer <key>`
   - `header` → `<HeaderName>: <Prefix><key>`
   - `query` → URL-encode `<ParamName>=<key>` and merge into the request URL (preserving existing query parameters)
   - `basic` → `Authorization: Basic <base64(username:key)>` (system performs the encoding; user is never asked for the encoded form)

### Wire-up

Extend `internal/core/connection_proxy.go:18-43` with the api-key dispatch branch:

```go
if _, ok := auth.Inner().(*config.AuthApiKey); ok {
    return apiKeyFactory.NewApiKey(c)
}
```

### Side benefit

`proxy_http` probes against api-key connectors begin working (today they hit `ErrProxyNotImplemented`). This is a prerequisite for the probe-runtime sub-issue to actually validate api-key credential health.

## Test plan

- [ ] Each placement produces the correct `http.Request` (header values, query string, encoding)
- [ ] `basic` correctly base64-encodes `username:key`
- [ ] `query` placement preserves existing query parameters when merging
- [ ] `header` placement honors `prefix` (with and without trailing space)
- [ ] `proxy_http` probe runs end-to-end against a stub upstream

## Dependencies

- AuthApiKey schema sub-issue
- `api_key_credentials` storage sub-issue

### PR DESCRIPTION
**Pull request by rmorlok**
Closes #253. Part of #243.

## Summary
New \`internal/auth_methods/api_key/\` package implementing \`iface.Proxy\`. With #250–#252 already merged, api-key connections can be created end-to-end but couldn't actually proxy outbound requests. This closes that gap.

Per-request flow:
1. Load active \`api_key_credentials\` row for the connection.
2. Decrypt the credentials blob (\`ApiKeyCredentialPlaintext\` JSON).
3. Apply the \`placement_snapshot\` stored on the row (or fall back to the live connector definition if no snapshot — defensive for rows that predate snapshot capture).
4. Inject per placement:
   - \`bearer\` → \`Authorization: Bearer <key>\`
   - \`header\` → \`<HeaderName>: <Prefix><key>\`
   - \`query\` → \`<ParamName>=<key>\` appended to URL (existing params preserved)
   - \`basic\` → \`Authorization: Basic <base64(username:key)>\` (RFC 7617)

## Design note: pure-data inject helper
The credential→auth mapping is \`computeAuthApplication\`, which returns the (header name, header value) + (query name, query value) pair as plain data. The proxy applies the result via gentleman's \`SetHeader\` / \`SetQuery\`. This keeps unit tests fast and assertable without booting an HTTP transport — \`inject_test.go\` covers every placement and error branch directly against the return value, and \`proxy_test.go\` covers the surrounding \`resolveAuth\` (DB read + decrypt + placement resolution).

## Wire-up
- \`internal/core/service_api_key.go\`: lazy factory accessor mirroring \`getOAuth2Factory\`. One factory per service, shared across connections.
- \`internal/core/connection_proxy.go\`: dispatch branch for \`AuthApiKey\`.
- \`internal/core/service.go\`: factory field + once guard.

## Side benefit
\`proxy_http\` probes against api-key connectors now run end-to-end (previously \`ErrProxyNotImplemented\`). Prerequisite for #255 to actually validate api-key health.

## Mock change
\`internal/core/mock/connection.go\` gains a \`ConnectorVersionEntity\` field so the resolveAuth fallback test can inject a live connector definition. Default behaviour (nil) is unchanged.

## Test plan
- [x] Bearer, header (with and without prefix), query, basic — \`computeAuthApplication\` unit tests.
- [x] Basic correctly base64-encodes \`username:key\` per RFC 7617, including the colon-in-username edge case.
- [x] Query value passes through verbatim (URL encoding is the transport layer's job).
- [x] Header honors prefix (with trailing space, with non-space separator like \`v1_\`).
- [x] Missing required fields surface errors: nil placement, empty api_key, missing header_name / param_name / username.
- [x] \`resolveAuth\` happy path with placement_snapshot.
- [x] \`resolveAuth\` falls back to live connector definition when snapshot is nil.
- [x] DB error from \`GetActiveApiKeyCredential\` propagates.
- [x] Malformed plaintext (non-JSON) surfaces \`unmarshal\` error.
- [x] Basic placement end-to-end resolves both username and key from the decrypted plaintext.
- [x] Full module green on SQLite + Postgres.
- [x] \`./scripts/preflight.sh\` green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/auth_methods/api_key/proxy.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+func (a *apiKeyConnection) resolveAuth(ctx context.Context) (authApplication, error) {
+	cred, err := a.db.GetActiveApiKeyCredential(ctx, a.connection.GetId())
+	if err != nil {
+		return authApplication{}, fmt.Errorf("failed to load api-key credential: %w", err)
+	}
+
+	placement := cred.PlacementSnapshot
+	if placement == nil {
+		// Fall back to the connector definition's placement when the credential
+		// row didn't snapshot one (rows inserted before snapshot wiring shipped).
```
Comment (rmorlok) [inline internal/auth_methods/api_key/proxy.go]: You do not need to support cases before this was added. Assume no production deploys for now.

Comment (rmorlok) [reply internal/auth_methods/api_key/proxy.go]: Dropped in c761fc7. \`resolveAuth\` now requires \`PlacementSnapshot\` and surfaces an error if it's missing — no fallback, no live-definition lookup. Also reverted the \`mock/connection.go\` \`ConnectorVersionEntity\` field that was added only to support the fallback test, since it's no longer needed.

Net: \`proxy.go\` -16 lines, \`mock/connection.go\` back to its prior shape.
